clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -730,7 +730,12 @@ getKernelInvocationKind(FunctionDecl *KernelCallerFunc) {
 }
 
 static const CXXRecordDecl *getKernelObjectType(FunctionDecl *Caller) {
-  return (*Caller->param_begin())->getType()->getAsCXXRecordDecl();
+  auto KernelParamTy = (*Caller->param_begin())->getType();
```

Comment:
Don't use auto here, according to the LLVM coding guidelines.

Reply:
Got it! Done.